cluster-autoscaler/cloudprovider/oci/oke/oci/oci_ref.go

```diff
@@ -0,0 +1,34 @@
+/*
+Copyright 2020-2023 Oracle and/or its affiliates.
+*/
+
+package oci
+
+import (
+	"strings"
+
+	apiv1 "k8s.io/api/core/v1"
+)
+
+// OciRef contains s reference to some entity in OCI world.
+type OciRef struct {
+	CompartmentID string
+	NodePoolID    string
+	InstanceID    string
+}
+
+func nodeToOciRef(n *apiv1.Node) (OciRef, error) {
+	return OciRef{
+		// This field comes from the CCM.
+		CompartmentID: n.Annotations["oci.oraclecloud.com/compartment-id"],
```

Comment:
This function and even this annotation string are both examples of code-reuse opportunities. 

`const ociAnnotationCompartmentID string = "oci.oraclecloud.com/compartment-id"` is is defined in `oci_cloud_provider.go` and the now two separate implementations of `nodeToOciRef` should be combined into one shared function.

Reply:
completed with current refactor